src/systems/systems/src/electrical/emergency_generator.rs

```diff
@@ -121,6 +121,8 @@ impl SimulationElement for EmergencyGenerator {
 
 #[cfg(test)]
 mod emergency_generator_tests {
+    use std::time::Duration;
```

Comment:
**thought**: I would expect some tests to be added or changed for the `EmergencyGenerator` changes. As I wrote all those tests before, I doubt they are perfect! 😀

Reply:
They are perfect, they were failing when updating power demand :D 

There's not much complexity on electrical side I think as long as it provides electricity when asked too. Troubles comes with hydraulic interaction so hope this is covered on hydraulics tests side. Will see if I have ideas of tests to add here